src/components/TodoItem.tsx

```diff
@@ -0,0 +1,152 @@
+import { useContext, useEffect, useState } from 'react';
+import classNames from 'classnames';
+import { Todo } from '../types/Todo';
+
+import { TodoContext } from './TodosContext';
+import { updateTodo } from '../api/todos';
+
+interface Props {
+  todo: Todo;
+}
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+}) => {
+  const {
+    toggleCompleted,
+    deleteTodo,
+    processingIds,
+    setErrorMessage,
+    setProcessingIds,
+    setTodos,
+    inputRef,
+  } = useContext(TodoContext);
+  const [isEditing, setEditing] = useState(false);
+  const [title, setTitle] = useState(todo.title);
+
+  function renameTodo(todoToUpdate: Todo, newTitle: string) {
+    setErrorMessage('');
+    setProcessingIds((current) => [...current, todoToUpdate.id]);
+
+    updateTodo({
+      ...todoToUpdate,
+      title: newTitle,
+    })
+      .then((updatedTodo) => {
+        setTodos((current) => current
+          .map((todoItem) => (todoItem.id === updatedTodo.id
+            ? updatedTodo
+            : todoItem)));
+        setEditing(false);
+      })
+      .catch(() => {
+        setErrorMessage('Unable to update a todo');
+        inputRef.current?.focus();
+      })
+      .finally(() => {
+        setProcessingIds((current) => current
+          .filter((id) => id !== todoToUpdate.id));
+      });
+  }
+
+  const handleToggleTodo = () => {
+    toggleCompleted(todo.id);
+  };
+
+  const handleDeleteTodo = () => {
+    deleteTodo(todo.id);
+  };
+
+  const handleEditingTodo = () => {
+    setEditing(true);
+    setTitle(todo.title.trim());
+  };
+
+  function handleSubmit(event: React.FormEvent) {
+    event.preventDefault();
+    const trimmedTitle = title.trim();
+
+    if (trimmedTitle === todo.title.trim()) {
+      setEditing(false);
+
+      return;
+    }
+
+    if (trimmedTitle === '') {
+      handleDeleteTodo();
+
+      return;
+    }
+
+    renameTodo(todo, trimmedTitle);
+  }
+
+  useEffect(() => {
+    if (isEditing && inputRef.current) {
+      inputRef.current.focus();
+    }
+  }, [isEditing, inputRef]);
+
+  return (
+    <div data-cy="Todo" className={todo.completed ? 'todo completed' : 'todo'}>
+      <label className="todo__status-label">
+        <input
+          data-cy="TodoStatus"
+          type="checkbox"
+          className="todo__status"
+          checked={todo.completed}
+          onChange={handleToggleTodo}
+        />
+      </label>
+
+      {isEditing ? (
```

Comment:
```suggestion
      {!!isEditing ? (
```
in rendering with &&  need to convert the value to boolean.